src/data-source/collection-browser-data-source.ts

```diff
@@ -0,0 +1,1189 @@
+import type { ReactiveController, ReactiveControllerHost } from 'lit';
+import {
+  AccountExtraInfo,
+  Aggregation,
+  Bucket,
+  CollectionExtraInfo,
+  FilterConstraint,
+  FilterMap,
+  FilterMapBuilder,
+  PageElementMap,
+  SearchParams,
+  SearchResult,
+  SearchType,
+} from '@internetarchive/search-service';
+import type { MediaType } from '@internetarchive/field-parsers';
+import {
+  prefixFilterAggregationKeys,
+  type FacetBucket,
+  type PrefixFilterType,
+  type TileModel,
+  PrefixFilterCounts,
+} from '../models';
+import type {
+  CollectionBrowserSearchInterface,
+  CollectionTitles,
+  PageSpecifierParams,
+} from './models';
+import { sha1 } from '../utils/sha1';
+
+type RequestKind = 'full' | 'hits' | 'aggregations';
+
+export interface CollectionBrowserDataSourceInterface
```

Comment:
suggestion: move the interface to another tile to reduce this file's line length